content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,520 @@
+1. Chi-Square Test
+2. T-Test
+3. ANOVA Test
+ 
+### Chi-square test - Python implementation.
+ 
+Chi-square test is used to perform testing on two categorical variables in our data population. It is focused on looking for an important relationship between two categorical variables.
+For example, comparing the aftereffects of malaria drug A, malaria drug B, and malaria drug C.
+Chi-square is a tool for checking how divergent the observed frequency is from the expected frequency.
+With Chi-Square, we accept the null hypothesis when variable A and variable B are not dependent, i.e. there is no relationship between them.
+We reject the null hypothesis(or accept the alternate hypothesis) when variable A depends on variable B, i.e. there is a relationship between them. The formula for calculating chi-square test is:
+ 
+![chi_square_formula](/engineering-education/statistical-hypothesis-testing-python-implementation/chi-square.PNG)
+ 
+Consider the following example:
+ 
+A mathematics teacher conducted a test for a class of 20 students. He expected 5 to score good marks, 7 to score average marks while 8 to fail. Eventually, when the result came out, 5 failed, 6 were average while 9 scored good marks. Mathematically, carry out chi-square test for this class.
+ 
+|  | Good  |  Average  | Fail | Total |
+|----|--------|:----------|--------------|--------|
+| Observed| 9 | 6 | 5 | 20 |
+| Expected| 5 | 7 | 8 | 20 |
+ 
+Where
+ 
+- observed good is A,
+- expected good is a,
+- observed average is B,
+- expected average is b,
+- observed fail is C, and
+- expected to fail is c.
+```
+Chi-square = Σ[(observed value - expected value)² / expected value]
+```
+```
+A-a = 4
+B-b = -1
+C-c = -3
+```
+```
+# find the square of our values
+/A-a/² = 16
+/B-b/² = 1
+/C-c/² = 9
+```
+```
+Good = (16/5) = 3.2
+Average = (1/7) = 0.143
+Fail = (9/8) = 1.125
+ 
+chi-square = 3.2 + 0.143 + 1.125
+chi-square = 4.468
+```
+For huge datasets, mathematical calculations cannot be done by hand. The implementation of the chi-square test in Python is discussed in the next section.
+ 
+```python
+import pandas as pd
+import numpy as np
+import scipy.stats as stats
+from scipy.stats import f_oneway
+from scipy.stats import chi2_contingency
+import matplotlib.pyplot as plt
+ 
+import statsmodels.api as sm
+from statsmodels.formula.api import ols
+ 
+import warnings
+warnings.filterwarnings( "ignore" )
+```
+ 
+The libraries imported are the most important libraries used for any data science python implementation work, except for the f_oneway used for the implementing one way ANOVA test and chi2_contigency used for the implementation of Chi-square.
+ 
+I will be using the popular iris data set for the tutorial, you can get it almost anywhere but I downloaded this from [Kaggle](https://www.kaggle.com/vikrishnan/iris-dataset).
+ 
+Load the data into the notebook
+```python
+df = pd.read_csv("/content/drive/MyDrive/IRIS.csv")
+```
+```python
+df.head()
+```
```

Comment:
The dataset, when downloaded does not have column names in it.
So do add a piece of code to assign column names to each column to prevent errors in the rest of the program.